default_dial_option_server_option_test.go

```diff
@@ -58,6 +58,24 @@ func (s) TestAddGlobalDialOptions(t *testing.T) {
 	}
 }
 
+// TestDisableGlobalOptions tests dialing with the disableGlobalDialOptions dial
+// option. Dialing with this set should not pick up global options.
+func (s) TestDisableGlobalOptions(t *testing.T) {
+	// Set transport credentials as a global option.
+	internal.AddGlobalDialOptions.(func(opt ...DialOption))(WithTransportCredentials(insecure.NewCredentials()))
+	// Dial with the disable global options dial option. This dial should fail
+	// due to the global dial options with credentials not being picked up due
+	// to global options being disabled.
+	if _, err := Dial("fake", internal.DisableGlobalDialOptions.(func() DialOption)()); err == nil {
+		t.Fatalf("Dialing without a credential did not fail")
+	} else {
+		if !strings.Contains(err.Error(), "no transport security set") {
```

Comment:
Nit: combine these lines.  You can also combine both cases themselves by doing:

```go
if _, err := Dial("fake", internal.Disable...); !strings.Contains(fmt.Sprint(err), "no transport security set") {
```

Reply:
Oh right nice thanks. Switched.